StripeConnect/StripeConnectTests/Internal/Webview/ConnectComponentWebViewControllerTests.swift

```diff
@@ -222,13 +222,9 @@ class ConnectComponentWebViewControllerTests: XCTestCase {
 
         try await webVC.webView.evaluateOnLoaderStart(elementTagName: "payouts")
 
-        if !webVC.activityIndicator.isAnimating {
-            XCTAssertFalse(webVC.activityIndicator.isAnimating)
-        } else {
-            // If it is animating, wait a second before we check.
-            sleep(1)
-            XCTAssertFalse(webVC.activityIndicator.isAnimating)
-        }
+        // Wait for the animation state to settle after the async operation
+        await Task.sleep(nanoseconds: 100_000_000) // 100ms
+        XCTAssertFalse(webVC.activityIndicator.isAnimating)
```

Comment:
Initially there was no if/else, and was only:
```
            XCTAssertFalse(webVC.activityIndicator.isAnimating)
```

I tried to de-flake it with the if/else statement you see there. Your proposed approach seems to introduce a less generous (100ms) sleep, which seems less likely to work?

Reply:
I think we need to wait for the JS to finish executing so that it can tell the activity indicator to stop animating, 100ms after executing the JS on the loaded page seems like plenty of time for that to happen. (And it seems to work, testing it here.) But when we were calling sleep(), I don't think the webview was able to message the activity indicator to stop in time.

The if statement was also confusing to me, because we're checking if it's false and then immediately asserting that it's false?

Reply:
Ah! Blocking the main thread -- maybe? in that case, yea.. lets try. try await

True.   Guess if it was false, there was no need to assert.